queries/nim/highlights.scm

```diff
@@ -0,0 +1,797 @@
+; SPDX-FileCopyrightText: 2023 Leorize <leorize+oss@disroot.org>, aMOPel <>
+; SPDX-License-Identifier: MPL-2.0
+; SPDX-License-Identifier: Apache-2.0
+
+; =============================================================================
+; catch all rules
+
+((identifier) @variable (#set! "priority" 95))
```

Comment:
Is the priority necessary?

Reply:
this priority is necessary, because the capture is most specific 
and will overrule things like (type_expression) which contain (identifier)s

I'm gonna put this in a comment

Reply:
Then we can move `(identifier)` up, and then add a `(type_expression (identifier) @type)` instead of needing priorities

Reply:
`@variable` capture is already at the top.

I had it wrong in my mind. the situation is this:

1. since `(type_expression)`s can infinitely nest, we have to capture the `(identifier)`s inside as `@type` with `has-ancestor` directive,
2. to have `@variable` captures not override `@type` captures those need to come before
3. NOW to have `@type` capture not override `@parameter` captures inside `(proc_type)`s, `@type` captures need to be before `@parameter`  captures.

But you're right, I can remove both priorities and push the `@parameter` block behind the `@type` block